### ISSUE TITLE
Allow full URL database name

### ISSUE THREAD

**Issue by timmillwood** [doctrine/couchdb-client#47]
I am working on a Drupal implementation of the CouchDB API and we have this working (pretty much) perfectly with the native CouchDB replicator and PouchDB, but the CouchDB Client doesn't seem to like that we have namespaced our endpoints. For example: example.com/relaxed/{db}.

I would like to propose we allow `CouchDBClient::create()` to accept a full URL or just the database name in the `dbname` option.

In the native client we pass in the values as, for example:
```
{"source": "http://admin:admin@drupal.loc/relaxed/default", "target": "http://localhost:5984/target", "worker_processes": 1}
```
but
```
{"source": "http://admin:admin@drupal.loc/relaxed/default", "target": "target", "worker_processes": 1}
```
would also work. This could be similar in CouchDB client.

I am however open to other solutions, but this seems the most straightforward approach.

**Comment by beberlei** [comment]
How about we don't use `dbname` for this but a new parameter `url`. If that is present everything else is ignored and we use `parse_url` to generate the old way of configuring things.

This is something we have been doing in DBAL as well.

**Comment by beberlei** [comment]
Implemented in 4812a4d
